app/v2/connectors/RetrieveUkPropertyPeriodSummaryConnector.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.connectors
+
+import config.AppConfig
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import v2.connectors.httpparsers.StandardIfsHttpParser._
+import v2.models.request.retrieveUkPropertyPeriodSummary.RetrieveUkPropertyPeriodSummaryRequest
+import v2.models.response.retrieveUkPropertyPeriodSummary.RetrieveUkPropertyPeriodSummaryResponse
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RetrieveUkPropertyPeriodSummaryConnector @Inject()(val http: HttpClient,
+                                                         val appConfig: AppConfig) extends BaseIfsConnector {
+
+  def retrieveUkProperty(request: RetrieveUkPropertyPeriodSummaryRequest)(
```

Comment:
You need to be able to detect when a foreign property comes back as the downstream endpoint is shared. See how this is done in [RetrieveUkPropertyAnnualSubmissionConnector.scala](https://github.com/hmrc/property-business-api/blob/main/app/v2/connectors/RetrieveUkPropertyAnnualSubmissionConnector.scala) which returns a richer result in order to signal this back to the service which can in turn issue a RULE_TYPE_OF_BUSINESS_INCORRECT if necessary.

You will need changes to the service and connector specs too to cover these various scenarios.